tsdb/shard.go

```diff
@@ -460,11 +460,23 @@ func (s *Shard) ready() error {
 
 // LastModified returns the time when this shard was last modified.
 func (s *Shard) LastModified() time.Time {
-	engine, err := s.Engine()
+	t, err := s.LastModifiedOrErr()
 	if err != nil {
 		return time.Time{}
 	}
-	return engine.LastModified()
+
+	return t
+}
+
+// LastModifiedOrErr returns the time when this shard was last modified or an error.
+// See: https://github.com/influxdata/plutonium/pull/4275#discussion_r2214077374
+// Previously we would throw away any error from Shard.LastModified
+func (s *Shard) LastModifiedOrErr() (time.Time, error) {
```

Comment:
Pedantry alert! `LastModifiedOrErr` always returns the last modified time _and_ the error. I think it should be `LastModifiedAndErr` or `LastModifiedWithErr`.